writers/writer.go

```diff
@@ -5,35 +5,43 @@ package writers
 
 import (
 	"fmt"
-
+	"github.com/BurntSushi/toml"
 	"github.com/gogo/protobuf/proto"
 	"github.com/mainflux/mainflux"
-	log "github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/errors"
+	"github.com/mainflux/mainflux/logger"
 	"github.com/mainflux/mainflux/transformers"
 	"github.com/mainflux/mainflux/transformers/senml"
 	nats "github.com/nats-io/nats.go"
+	"io/ioutil"
+)
+
+var (
+	errOpenConfFile = errors.New("Unable to open configuration file")
+	errParseConfFile = errors.New("Unable to parse configuration file")
 )
 
 type consumer struct {
 	nc          *nats.Conn
-	channels    map[string]bool
+	subjects    []string
 	repo        MessageRepository
 	transformer transformers.Transformer
-	logger      log.Logger
+	logger      logger.Logger
 }
 
 // Start method starts consuming messages received from NATS.
 // This method transforms messages to SenML format before
 // using MessageRepository to store them.
-func Start(nc *nats.Conn, repo MessageRepository, transformer transformers.Transformer, queue string, channels map[string]bool, logger log.Logger) error {
+func Start(nc *nats.Conn, repo MessageRepository, transformer transformers.Transformer, queue string, filters FiltersCfg, logger logger.Logger) error {
 	c := consumer{
 		nc:          nc,
-		channels:    channels,
+		subjects:    filters.subjects,
 		repo:        repo,
 		transformer: transformer,
 		logger:      logger,
 	}
 
+	// TODO subscribe with only selected subjects
```

Comment:
I think that here you can pass `subjectsConfigPath` as argument and  call `LoadSubjectsConfig` to avoid the duplication in all writers.